scripts/transferIssues.js

```diff
@@ -0,0 +1,415 @@
+  const packages = [
+    ...(await fs.readdir('packages')),
+    'magi-cli', // for testing only
+    'web-components' // for testing only
+  ];
+
+  const repos = await Promise.all(
+    packages
+      .filter((package) => {
+        const skip = EXCLUDED_PACKAGES.indexOf(package) > -1;
+        if (skip) {
+          console.log(`Skipped the package ${package} as it's in the skip list`);
+        }
+        return !skip;
+      })
+      .map(async (package) => {
+        try {
+          const { data: repo } = await octokit.rest.repos.get({
+            owner: 'vaadin',
+            repo: package
+          });
+          return repo;
+        } catch (e) {
+          if (!!e.constructor && e.constructor.name === 'RequestError' && e.status === 404) {
+            console.log(`Skipped the package ${package} as it has no separate repo`);
+            return;
+          } else {
+            throw e;
+          }
+        }
+      })
+  );
+
+  return repos
+    .filter((repo) => !!repo) // remove skipped packages from the list
+    .filter((repo) => {
+      const skip = shouldExcludeRepo(repo);
+      if (skip) {
+        console.log(`Skipped the repo ${repo.full_name} because the shouldExcludeRepo() filter returned true`);
+      }
+      return !skip;
+    });
+}
+
+const zhWorkspaceNameById = new Map();
+async function getZenHubWorkspaceName(workspace_id, repo_id) {
+  if (!zhWorkspaceNameById.has(workspace_id)) {
+    zhWorkspaceNameById.set(
+      workspace_id,
+      (async (workspace_id, repo_id) => {
+        const { data: workspaces } = await zhApi.get(`/p2/repositories/${repo_id}/workspaces`);
+        const idx = workspaces.findIndex((workspace) => workspace.id === workspace_id);
+        if (idx > -1) {
+          return workspaces[idx].name;
+        } else {
+          throw new Error(`Cannot find ZenHub workspace with ID ${workspace_id} for the repo with ID ${repo_id}`);
+        }
+      })(workspace_id, repo_id)
+    );
+  }
+
+  return zhWorkspaceNameById.get(workspace_id);
+}
+
+// Special handling for ZenHub REST API rate limiting
+// (see https://github.com/ZenHubIO/API#api-rate-limit)
+function zhRateLimitingAdapter(adapter) {
+  return async (config) => {
+    let response;
+    let status = 403;
+    while (status === 403) {
+      try {
+        response = await adapter(config);
+        status = response.status;
+      } catch (e) {
+        if (e.isAxiosError && e.response.status === 403) {
+          const resetAtMs = e.response.headers['x-ratelimit-reset'] * 1000;
+          const sentAtMs = new Date(e.response.headers['date']).getTime();
+          const timeoutMs = resetAtMs - sentAtMs;
+          console.log(`timeout until ZenHub API request rate reset: ${timeoutMs} ms`);
+          await new Promise((r) => setTimeout(r, timeoutMs));
+        } else {
+          throw e;
+        }
+      }
+    }
+
+    return response;
+  };
+}
+
+async function createLabelInRepo(label, repo) {
+  let newLabel;
```

Comment:
not blocking, would it be good to mark the transferred ticket with the original component, so that it is easy to recognize where the ticket comes?

Reply:
Yes! That's a very good point!
I remember thinking of this, and yet I've totally missed it in the end.